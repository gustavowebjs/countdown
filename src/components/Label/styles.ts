import styled from 'styled-components';

export const Container = styled.div`
  margin-bottom: 10px;
`;

export const Label = styled.label`
  color: ${({ theme }) => theme.colors.neutral.dark};
`;
