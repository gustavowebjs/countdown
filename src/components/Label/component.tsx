import { ReactElement } from 'react';
import * as T from './types';
import * as S from './styles';

const Label = (props: T.LabelProps): ReactElement => {
  const { text } = props;

  return (
    <S.Container>
      <S.Label>{text}</S.Label>
    </S.Container>
  );
};

export default Label;
