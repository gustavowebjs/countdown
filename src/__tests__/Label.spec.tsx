import { shallow } from 'enzyme';
import Label from '../components/Label';

describe('Label Component', () => {
  it('Should Label component render correctly', () => {
    const wrapper = shallow(<Label text="Label" />);
    expect(wrapper.text()).toEqual('Label');
    expect(wrapper).toMatchSnapshot();
  });
});
