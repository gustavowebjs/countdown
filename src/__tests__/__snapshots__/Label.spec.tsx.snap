// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component Should Label component render correctly 1`] = `
<styled.div>
  <styled.label>
    Label
  </styled.label>
</styled.div>
`;
